src/Symfony/Component/Routing/Matcher/RedirectableUrlMatcher.php

```diff
@@ -32,9 +32,9 @@ public function match($pathinfo)
             }
 
             try {
-                parent::match($pathinfo.'/');
+                $parameters = parent::match($pathinfo.'/');
 
-                return $this->redirect($pathinfo.'/', null);
+                return array_replace($parameters, $this->redirect($pathinfo.'/', isset($parameters['_route']) ? $parameters['_route'] : null));
```

Comment:
Passing the _route to the redirect method makes the tests fail in the FrameworkBundle: https://github.com/symfony/symfony/pull/19037/files#diff-7b7153654c7ee1d563e8421f3f9be20e

I could extract this fix as a bugfix as the current behavior is not consistent with the dumped matcher, see https://github.com/symfony/symfony/pull/19037/files#diff-3b72491a9ba1cff58442b845ae837eb3L292 which passed the route name as second argument to redirect.

But I don't know how to fix the frameworkbundle test which will also be run against the lowest allowed routing version, right? So I need to increase the routing dependency in https://github.com/symfony/symfony/blob/2.7/src/Symfony/Bundle/FrameworkBundle/composer.json#L28 but to a version that is not released yet?